cmd/stats/stats.go

```diff
@@ -185,19 +197,185 @@ func argsToColumnFilter(cmd *cobra.Command) []string {
 	}
 }
 
-// TODO will be implemented in #65
-func printStats(reps []Rep, cols []string) {
+func getStatsHeader(name string, lang string, start string, end string) (header string) {
+	nameSection := ""
+	if name != "" {
+		nameSection = name
+	} else if lang != "" {
+		// TODO make a map of languages to human readable names
+		nameSection = lang
+	}
+
+	dateSection := "from today"
+	if start != "" {
+		if end != "" {
+			dateSection = fmt.Sprintf("from %s to %s", start, end)
+		} else {
+			dateSection = fmt.Sprintf("since %s", start)
+		}
+	}
+
+	tmplData := struct {
+		// name of the file or language of the file
+		Name string
+		// "since X days", or "from YYYY-MM-DD to YYYY-MM-DD", etc
+		Date string
+	}{
+		Name: nameSection,
+		Date: dateSection,
+	}
+
+	tmplString := "" +
+		`stats {{if .Name}}for {{.Name}} {{end}}{{.Date}}:`
+
+	tmpl := template.Must(template.New("header").Parse(tmplString))
+	var buf bytes.Buffer
+	err := tmpl.Execute(&buf, tmplData)
+	if err != nil {
+		panic(err)
+	}
+	return buf.String()
+}
+
+// Calculates the average, min, max, first, last, and delta of a stat.
+// Returns that information in an array of strings.
+// Formats the data based on the column name selected.
+func getColumnStats(reps []Rep, colName string) []string {
+	getCol := func(r Rep, colName string) float64 {
+		switch colName {
+		case WPM:
+			return r.Wpm
+		case RAW_WPM:
+			return r.Raw
+		case DURATION:
+			return float64(r.Dur)
+		case ACCURACY:
+			return r.Acc
+		case MISTAKES:
+			return float64(r.Miss)
+		case UNCORRECTED_ERRORS:
+			return float64(r.Errs)
+		}
+		return -math.MaxFloat64
+	}
+
+	avg := 0.0
+	min := math.MaxFloat64
+	max := -math.MaxFloat64
+	first := getCol(reps[len(reps)-1], colName)
+	last := getCol(reps[0], colName)
+	delta := last - first
+
+	sum := 0.0
+	for _, rep := range reps {
+		curr := getCol(rep, colName)
+		sum += curr
+		if curr < min {
+			min = curr
+		}
+		if curr > max {
+			max = curr
+		}
+	}
+	avg = sum / float64(len(reps))
+
+	colData := []float64{avg, min, max, first, last, delta}
+
+	row := []string{}
+
+	for _, d := range colData {
+		row = append(row, util.ColumnString(colName, d))
+	}
+	return row
+}
+
+func printStats(cmd *cobra.Command, reps []Rep) {
```

Comment:
This name is misleading. The stats table is the one that contains the calculations of avg, min, max, and so on. The graph is printed below that. The rows are printed below the graph.

Change this name to `render` instead.